packages/astro/src/vite-plugin-astro-server/plugin.ts

```diff
@@ -70,3 +78,29 @@ export default function createVitePluginAstroServer({
 		},
 	};
 }
+
+/**
+ * It creates a `SSRManifest` from the `AstroSettings`.
+ *
+ * Renderers needs to be pulled out from the page module emitted during the build.
+ * @param settings
+ * @param renderers
+ */
+export function generateDevelopmentManifest(settings: AstroSettings): SSRManifest {
```

Comment:
I'm curious to see how this would be used later. IIUC the `SSRManifest` is only used to pass info from build to the SSR runtime, where it has the special restriction of implementing [serialization](https://github.com/withastro/astro/blob/ab922c7f8e3fc367a3ff0f7921dbe01c54c58985/packages/astro/src/core/build/plugins/plugin-ssr.ts#L475) and [deserialization](https://github.com/withastro/astro/blob/ab922c7f8e3fc367a3ff0f7921dbe01c54c58985/packages/astro/src/core/app/common.ts#L4-L27) 🤔

Reply:
I also like that idea. On my initial attempts, I tried to create a `renderRoute` too, but it turns out we need to pass in a lot of options. Right now there seem to be 3 key things needed to render a page: `mod` (the runtime module instance), `RenderContext` (created per request), and `Environment` (created once).

IIUC it would be nice to lessen these 3 things. Maybe we can discuss more about your plans with involving `SSRManifest` into the mix. Maybe it'll merge into one of the 3 things?